src/main/java/uk/gov/pay/publicauth/resources/PublicAuthResource.java

```diff
@@ -144,12 +154,14 @@ public Response updateTokenDescription(JsonNode payload) throws ValidationExcept
     @DELETE
     public Response revokeSingleToken(@PathParam("accountId") String accountId, JsonNode payload) throws ValidationException, TokenNotFoundException {
 
-        validatePayloadHasFields(payload, Collections.emptyList(), asList(TOKEN_LINK_FIELD, TOKEN_HASH_FIELD));
+        validatePayloadHasFields(payload, Collections.emptyList(), asList(TOKEN_LINK_FIELD, TOKEN_FIELD));
 
-        if (payload.hasNonNull(TOKEN_HASH_FIELD)) {
-            return authDao.revokeSingleToken(accountId, TokenHash.of(payload.get(TOKEN_HASH_FIELD).asText()))
-                    .map(this::buildRevokedTokenResponse)
-                    .orElseThrow(() -> new TokenNotFoundException("Could not revoke token"));
+        if (payload.hasNonNull(TOKEN_FIELD)) {
+             return tokenService.extractEncryptedTokenFrom(payload.get(TOKEN_FIELD).asText())
+            .map(token -> authDao.revokeSingleToken(accountId, TokenHash.of(token.getName()))
+                    .map(this::buildRevokedTokenResponse))
+                     .orElseThrow(() -> new TokenNotFoundException("Could not revoke token"))
+                     .orElseThrow(() -> new TokenNotFoundException("Could not retrieve token_hash from token"));
```

Comment:
This message doesn’t quite fit: there’s not a property called `token_hash` in `token`

Reply:
Righto